### PR TITLE
ResourceManager documentation

### PR DESCRIPTION
**Pull request by Sollace**
Documented the behaviour of all the methods, as well as some of the gotchas.

### CODE REVIEW

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 1]
```diff
@@ -1,13 +1,48 @@
 CLASS net/minecraft/class_3300 net/minecraft/resource/ResourceManager
+	COMMENT Provides resource loading capabilities to Minecraft.
 	METHOD method_14486 getResource (Lnet/minecraft/class_2960;)Lnet/minecraft/class_3298;
+		COMMENT Finds and returns the corresponding resource for a resource's identifier.
+		COMMENT <br>
+		COMMENT Starts by scanning each resourcepack from highest priority to lowest. If no resourcepacks were found
+		COMMENT to contain the requested entry, will throw a FileNotFoundException.
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT Starts by scanning each resource pack from highest priority to lowest. If no resource packs were found
		COMMENT to contain the requested entry, will throw a {@code FileNotFoundException}.
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+		COMMENT <br>
+		COMMENT @throws FileNotFoundException if the identified resource could not be found, or could not be loaded.
+		COMMENT @throws IOException if the identified resource was found but a stream to it could not be opened.
 		ARG 1 id
+			COMMENT The resource identifier to search for
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
			COMMENT the resource identifier to search for
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
 	METHOD method_14487 getAllNamespaces ()Ljava/util/Set;
+		COMMENT Gets a set of all namespaces offered by the resourcepacks loaded by this manager.
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT Gets a set of all namespaces offered by the resource packs loaded by this manager.
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 4]
```diff
@@ -1,13 +1,48 @@
...
 	METHOD method_14488 findResources (Ljava/lang/String;Ljava/util/function/Predicate;)Ljava/util/Collection;
-		ARG 1 resourceType
+		COMMENT Returns a sorted list of identifiers matching a path predicate.
+		COMMENT <br>
+		COMMENT Scanning begins in {@linkplain startingPath} and each candidate file present under that directory
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT Scanning begins in {@code startingPath} and each candidate file present under that directory
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 5]
```diff
@@ -1,13 +1,48 @@
...
+		COMMENT will be offered up to the predicate to decide whether it should be included or not.
+		COMMENT <br>
+		COMMENT Elements in the returned list may not, necessarily be unique. Additional effort is advised to ensure that
+		COMMENT duplicates in the returned list are discarded before loading.
+		COMMENT
+		COMMENT @return The list matching identifiers
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT @return the list matching identifiers
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 6]
```diff
@@ -1,13 +1,48 @@
...
+		ARG 1 startingPath
+			COMMENT The starting path to begin scanning from.
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
			COMMENT the starting path to begin scanning from
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 7]
```diff
@@ -1,13 +1,48 @@
...
 		ARG 2 pathPredicate
+			COMMENT A predicate to determine whether a path should be included or not.
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
			COMMENT a predicate to determine whether a path should be included or not
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 8]
```diff
@@ -1,13 +1,48 @@
...
 	METHOD method_14489 getAllResources (Lnet/minecraft/class_2960;)Ljava/util/List;
+		COMMENT Gets all of the available resources to the corresponding resource identifier.
+		COMMENT <br>
+		COMMENT Resources are returned in load order, or ascending order of priority, so the last element in the returned
+		COMMENT list is what would be returned normally by {@link getResource}
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT list is what would be returned normally by {@link #getResource}
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 9]
```diff
@@ -1,13 +1,48 @@
...
+		COMMENT <br>
+		COMMENT @throws FileNotFoundException if no matching resources could be found (i.e. if the list would be empty)
+		COMMENT @throws IOException if resources were found, but any one of them could not be opened to be read.
 		ARG 1 id
+			COMMENT The resource identifier to search for
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
			COMMENT the resource identifier to search for
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 10]
```diff
@@ -1,13 +1,48 @@
...
 	METHOD method_18234 containsResource (Lnet/minecraft/class_2960;)Z
+		COMMENT Checks whether any of the currently-loaded resourcepacks contain an entry for the given id.
+		COMMENT <br>
+		COMMENT Starts by querying the resourcepack with the highest priority to lowest until it finds one that
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT Checks whether any of the currently-loaded resource packs contain an entry for the given id.
		COMMENT <br>
		COMMENT Starts by querying the resource pack with the highest priority to lowest until it finds one that
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 11]
```diff
@@ -1,13 +1,48 @@
...
+		COMMENT responds to the requested identifier.
 		ARG 1 id
+			COMMENT The resource identifier to search for
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
			COMMENT the resource identifier to search for
```

File: mappings/net/minecraft/resource/ResourceManager.mapping [hunk 12]
```diff
@@ -1,13 +1,48 @@
...
 	METHOD method_29213 streamResourcePacks ()Ljava/util/stream/Stream;
+		COMMENT Gets a stream of loaded resourcepacks in increasing order of priority.
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceManager.mapping]:
```suggestion
		COMMENT Gets a stream of loaded resource packs in increasing order of priority.
```

**Review by YanisBft — APPROVED** [review]

**Review by Gegy — APPROVED** [review]

**Review by Juuxel — CHANGES_REQUESTED** [review]
`<p>` should be used instead of `<br>`. (Contents themselves look fine)

**Pull request merged**